static/scripts/index.js

```diff
@@ -0,0 +1,513 @@
+                            } else {
+                                if ((load_q1 === 1 && load_q2 === 0) || (load_q1 === 0 && load_q2 === 1)) {
+                                    TD.style.backgroundColor = 'green';
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            customBorders: [
+                {
+                    range: {
+                        from: {
+                            row: 0,
+                            col: 0,
+                        },
+                        to: {
+                            row: nbrLines,
+                            col: 6,
+                        },
+                    },
+                    end: {
+                        width: 2,
+                        color: 'black'
+                    },
+                },
+                {
+                    range: {
+                        from: {
+                            row: 0,
+                            col: 0,
+                        },
+                        to: {
+                            row: 5,
+                            col: nbrCols,
+                        },
+                    },
+                    bottom: {
+                        width: 2,
+                        color: 'black'
+                    },
+                },
+                {
+                    row: 5,
+                    col: 6,
+                    bottom: {
+                        width: 2,
+                        color: 'black'
+                    }
+                },
+            ],
+            beforeFilter(conditionsStack) {
+                const filtersPlugin = this.getPlugin('filters');
+                const tab = this.getData();
+
+                let values = [];
+                const filteredResults = [];
+                const col = conditionsStack[0].column;
+
+                if (conditionsStack && conditionsStack.length > 0) {
+                    for (let i = 0; i < conditionsStack.length; i++) {
+                        values = conditionsStack[i].conditions[0].args.flat();
+
+                        for (const row of tab) {
+                            if (values.includes(row[col])) {
+                                filteredResults.push(row[0]);
+                            }
+                        }
+                    }
+                }
+                filtersPlugin.clearConditions();
+                filtersPlugin.addCondition(0, 'by_value', [filteredResults]);
+            },
+            afterFilter(conditionsStack) {
+                const filtersPlugin = this.getPlugin('filters');
+                const filtersRowsMap = filtersPlugin.filtersRowsMap;
+
+                // Exclude fixed lines from the filter
+                for (let i = 0; i < lenFixedRowsText; i++) {
+                    filtersRowsMap.setValueAtIndex(i, false);
+                }
+            },
```

Comment:
Is this piece of code really used ? It seems a bit confusing to me.

Reply:
Yes, it's used to filter data. Handsontable does not take merged rows into account when sorting. So I have to do it manually, I retrieve which values are requested and in which column. Then I go through the rows of user data and check whether the value matches the one requested to retrieve the row id. Then I create a new condition, this time based on the id, to get the merged rows.